manifests/params.pp

```diff
@@ -29,9 +29,21 @@
   if $::operatingsystem == 'Ubuntu' {
     if versioncmp($::lsbdistrelease, '8.04') < 1 {
       $init_style = 'debian'
-    } else {
+    } elsif versioncmp($::operatingsystemrelease, '15.04') < 0 {
       $init_style = 'upstart'
     }
+    else {
+      # Introducing fact 'is_systemd' since Ubuntu 15.04 can be systemd OR upstart (defaults to systemd though)
+      # Fact allows us to retrieve the client side value.
+      # We convert to boolean incase some people may have stringify facts turned on or not set in puppet 3.x still.
+      $is_systemd = str2bool($::is_systemd)
```

Comment:
Ya, I saw that.  I'm pretty sure mine will work, theoretically it does :)  But I may call it a weekend before I get any thorough testing done on it and return to fixing it on Monday.  I'm hoping to deploy consul using this to our University soon enough so I want to make sure everything is up to snuff.  Since there is no yum packages or apt that I know of for consul in the distros that we're using.